### PR TITLE
Brms 76 better logs

### PR DESCRIPTION
**Pull request by manishsw**
created BRM logger to log request with time, id, audio source , api version.

### CODE REVIEW

File: app/uk/gov/hmrc/brm/controllers/BirthEventsController.scala [hunk 1]
```diff
@@ -50,16 +52,17 @@ trait BirthEventsController extends controller.BaseController with HeaderValidat
...
   private def handleException(method: String)(implicit payload: Payload): PartialFunction[Throwable, Result] = {
     case Upstream4xxResponse(message, NOT_FOUND, _, _) =>
       getMetrics().connectorStatus(NOT_FOUND)
-      Logger.warn(s"[BirthEventsController][Connector][$method] NotFound: $message")
+      warn(CLASS_NAME, "handleException", s"NotFound: $message.")
       respond(Ok(Json.toJson(BirthResponseBuilder.withNoMatch())))
     case Upstream4xxResponse(message, BAD_REQUEST, _, _) =>
       getMetrics().connectorStatus(BAD_REQUEST)
-      Logger.warn(s"[BirthEventsController][Connector][$method] BadRequest: $message")
+      warn(CLASS_NAME, "handleException", s"BadRequest: $message.")
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/controllers/BirthEventsController.scala]: Change this to pass in `$method`

File: app/uk/gov/hmrc/brm/utils/Keygenerator.scala [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+/**
+  * Created by user on 15/09/16.
+  */
+object Keygenerator {
+
+  val DATE_FORMAT: String = "yyyyMMdd:HHmmssSS"
+  private var keyForRequest: String = null
+  private val versionKey: String = "version"
+  val matchHeader: String => Option[Match] =
+    new Regex( """^application/vnd[.]{1}hmrc[.]{1}(.*?)[+]{1}(.*)$""", versionKey, "contenttype") findFirstMatchIn _
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/utils/Keygenerator.scala]: Can we refactor this into a common util?

File: app/uk/gov/hmrc/brm/utils/Keygenerator.scala [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  private def getDateKey: String = {
+    var dateTime = new DateTime()
+    val formatter = DateTimeFormat.forPattern(DATE_FORMAT);
+    val formattedDate: String = formatter.print(dateTime)
+    formattedDate
+  }
+
+  private def getApiVersion(request: Request[JsValue]): String = {
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/utils/Keygenerator.scala]: Remove this duplication and place this in the same util file.

**Review by adamconder — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/brm/controllers/BirthEventsController.scala (hunk 1 above)
Comment (manishsw) [reply app/uk/gov/hmrc/brm/controllers/BirthEventsController.scala]: done.

File: app/uk/gov/hmrc/brm/utils/Keygenerator.scala (hunk 2 above)
Comment (manishsw) [reply app/uk/gov/hmrc/brm/utils/Keygenerator.scala]: moved to common util.

File: app/uk/gov/hmrc/brm/utils/HeaderValidator.scala [hunk 4]
```diff
@@ -40,7 +41,7 @@ trait HeaderValidator extends Results {
 
   val matchAuditSource : String => Option[Match] = new Regex("""^(.*)$""", "auditsource") findFirstMatchIn _
 
-  val matchHeader : String => Option[Match] = new Regex( """^application/vnd[.]{1}hmrc[.]{1}(.*?)[+]{1}(.*)$""", "version", "contenttype") findFirstMatchIn _
+  //val matchHeader : String => Option[Match] = new Regex( """^application/vnd[.]{1}hmrc[.]{1}(.*?)[+]{1}(.*)$""", "version", "contenttype") findFirstMatchIn _
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/utils/HeaderValidator.scala]: Remove this code

**Review by adamconder — CHANGES_REQUESTED** [review]

**Review by adamconder — APPROVED** [review]

**Pull request merged**